File: api/backend/internal/model/dto/visit.go
package dto

import "time"

// ============ VisitController 请求 ============

// @request
// PageVisitByKeyQuery 分页查询来访登记请求
type PageVisitByKeyQuery struct {
	PageNum    *int    `json:"page_num" valid:"required"`  // 页码
	PageSize   *int    `json:"page_size" valid:"required"` // 条数
	ElderName  *string `json:"elder_name"`                 // 老人姓名
	VisitName  *string `json:"visit_name"`                 // 来访者姓名
	VisitPhone *string `json:"visit_phone"`                // 来访者电话
	VisitFlag  *string `json:"visit_flag"`                 // 来访状态
}

// @request
// EditVisitQuery 编辑来访登记请求
type EditVisitQuery struct {
	ID        *int64  `json:"id"`                          // id
	Name      *string `json:"name" valid:"required"`       // 来访者姓名
	Phone     *string `json:"phone" valid:"required"`      // 来访者电话
	Relation  *string `json:"relation" valid:"required"`   // 与老人关系
	VisitDate *string `json:"visit_date" valid:"required"` // 来访时间
	VisitNum  *int64  `json:"visit_num" valid:"required"`  // 来访者人数
}

// @request
// AddVisitQuery 新增来访登记请求（继承 EditVisitQuery）
type AddVisitQuery struct {
	EditVisitQuery
	ElderID *int64 `json:"elder_id" valid:"required"` // 老人编号
}

// @request
// RecordLeaveQuery 登记离开请求
type RecordLeaveQuery struct {
	ID        *int64  `json:"id" valid:"required"`         // id
	LeaveDate *string `json:"leave_date" valid:"required"` // 离开时间
}

// PageSearchElderByKeyQuery 分页搜索老人请求（定义见 elderrecord.go）

// ============ VisitController 响应 ============

// @response
// PageVisitByKeyVO 分页查询来访登记响应
type PageVisitByKeyVO struct {
	Rank
	ID         int64     `json:"id"`          // id
	ElderName  string    `json:"elder_name"`  // 老人姓名
	VisitName  string    `json:"visit_name"`  // 来访者姓名
	VisitPhone string    `json:"visit_phone"` // 来访者电话
	Relation   string    `json:"relation"`    // 与老人关系
	VisitDate  time.Time `json:"visit_date"`  // 来访时间
	LeaveDate  time.Time `json:"leave_date"`  // 离开时间
	VisitNum   int64     `json:"visit_num"`   // 来访者人数
	VisitFlag  string    `json:"visit_flag"`  // 来访状态
}

// @response
// GetVisitByIDVO 根据编号获取来访登记响应
type GetVisitByIDVO struct {
	ID         int64     `json:"id"`          // id
	ElderName  string    `json:"elder_name"`  // 老人姓名
	VisitName  string    `json:"visit_name"`  // 来访者姓名
	VisitPhone string    `json:"visit_phone"` // 来访者电话
	Relation   string    `json:"relation"`    // 与老人关系
	VisitDate  time.Time `json:"visit_date"`  // 来访时间
	VisitNum   int64     `json:"visit_num"`   // 来访者人数
}

File: mp-b/admin/api/visit.ts
/**
 * 来访登记相关接口
 * 对应后端 VisitController（/visit/...）
 */
import { get, post } from '../utils/request'
import { PageResult, mockPage } from './elder'

const PAGE_NUM = 1
const PAGE_SIZE = 200

export interface VisitItem {
	id: number
	elder_name: string
	visit_name: string
	visit_phone: string
	relation: string
	visit_date: string
	leave_date: string
	visit_num: number
	visit_flag: number
}

export interface VisitForm {
	id?: number
	elder_id: number
	name: string
	phone: string
	relation: string
	visit_date: string
	visit_num: number
}

// ====== MOCK 数据（后台未完成，测试前端用；后台完成后放开各函数内注释并删除本段）======
const mockVisitList: VisitItem[] = [
	{ id: 1, elder_name: '张建国', visit_name: '张小明', visit_phone: '13900000001', relation: '儿子', visit_date: '2026-08-20 10:00', leave_date: '2026-08-20 11:00', visit_num: 1, visit_flag: 1 },
	{ id: 2, elder_name: '李德福', visit_name: '李芳', visit_phone: '13900000009', relation: '女儿', visit_date: '2026-08-21 15:00', leave_date: '', visit_num: 2, visit_flag: 0 }
]

/** 来访列表（全量） */
export function pageVisitByKey(elderName?: string, visitFlag?: number): Promise<PageResult<VisitItem>> {
	// return get<PageResult<VisitItem>>('/visit/pageVisitByKey', {
	// 	page_num: PAGE_NUM,
	// 	page_size: PAGE_SIZE,
	// 	elder_name: elderName || undefined,
	// 	visit_flag: visitFlag != null ? '' + visitFlag : undefined
	// })
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	let list = mockVisitList
	if (elderName) list = list.filter((v) => v.elder_name.indexOf(elderName) >= 0)
	if (visitFlag != null) list = list.filter((v) => v.visit_flag === visitFlag)
	return Promise.resolve(mockPage(list))
}

export function getVisitById(id: number): Promise<VisitItem> {
	// return get<VisitItem>('/visit/getVisitById', { id })
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve(mockVisitList.find((v) => v.id === id) || mockVisitList[0])
}

export function addVisit(form: VisitForm): Promise<void> {
	// return post<void>('/visit/addVisit', form)
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve()
}

export function editVisit(form: VisitForm): Promise<void> {
	// return post<void>('/visit/editVisit', form)
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve()
}

export function recordLeave(id: number, leaveDate: string): Promise<void> {
	// return post<void>('/visit/recordLeave', { id, leave_date: leaveDate })
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve()
}

export function deleteVisit(id: number): Promise<void> {
	// return post<void>('/visit/deleteVisit', { id })
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve()
}
